proposals/0304-structured-concurrency.md

```diff
@@ -625,26 +665,126 @@ func detach<T>(
 Detached tasks will typically be created using a closure, e.g.,
 
 ```swift
-let dinnerHandle: Task.Handle<Meal, Error> = detach {
-  try await makeDinner()
-}
+let dinnerHandle: Task.Handle<Meal, Error> = 
+  asyncDetached {
+    try await makeDinner()
+  }
 
 try await eat(mealHandle: dinnerHandle)
 ```
 
-By default, the new task will be initially scheduled on the default global
-concurrent executor. Once custom executors are introduced in another proposal,
-these will be able to take an executor parameter to determine on which executor 
-to schedule the new task instead.
+By default, the new task will be initially scheduled on the default global concurrent executor. Once custom executors are introduced in another proposal, these will be able to take an executor parameter to determine on which executor to schedule the new task instead.
+
+##### Creating tasks in synchronous contexts using  `async {}`
+
+As discussed in the proposed solution section of the proposal, detached tasks are useful, but sacrifice a lot in the process of detaching from the enclosing context.
+
+We propose to introduce a new `async` function that addresses the above concerns and should be used when continuing the work of a synchronous function as async. It propagates both the priority and actor from where it is invoked into the closure, and suppresses the need for `self.`. Our example above will be rewritten as:
+
+```swift
+@MainActor func saveResults() { // synchronous function, inside the MainActor
+  view.startSavingSpinner()     // executes on the main actor, immediately
+  async { 
+    await ioActor.save()        // hop to ioActor to save
+    view.stopSavingSpinner()    // back on main actor to update UI
+  }
+}
+```
+
+The declaration of the `async` function is as follows:
+
+```swift
+@discardableResult
+func async(
+  operation: @Sendable @escaping () async -> Void
+) -> Task.Handle<Never, Never>
```

Comment:
@DougGregor @jckarter please sanity check this is what we want to be saying here.

To be honest I could see this having the same return type as detach... one could argue that If I'm able to wait on it i should use a child task anyway, but _technically_ it is not wrong or costly in any way to make this return the right type. One _could_ pass the handle to something that can await on it... although that's slowly moving into "weird, then just await on it" terrain so perhaps the Never returning is fine...?